### PR TITLE
Fixed Doctest for Bio.PDB

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1902 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This PR proposes changes to the following PDB files:
'Bio.PDB.Atom', No changes, no doctest on file.
'Bio.PDB.DSSP', Updated and changed to block test as it requires dssp
'Bio.PDB.Entity', No changes, no doctest on file.
'Bio.PDB.FragmentMapper', Updated
'Bio.PDB.mmcifio', Updated
'Bio.PDB.PDBIO', Updated
'Bio.PDB.ResidueDepth', Updated and changed to block test as it requires msms
'Bio.PDB.vectors', Updated, Note the comments suggest the result should be different but that is not a Doctest issue, code migth need review.

### CODE REVIEW

File: Bio/PDB/DSSP.py [hunk 1]
```diff
@@ -308,18 +312,20 @@ class DSSP(AbstractResiduePropertyMap):
...
-    >>> dssp = DSSP(model, "1mot.pdb")
-    >>> # DSSP data is accessed by a tuple (chain_id, res_id)
-    >>> a_key = list(dssp.keys())[2]
-    >>> # (dssp index, amino acid, secondary structure, relative ASA, phi, psi,
-    >>> # NH_O_1_relidx, NH_O_1_energy, O_NH_1_relidx, O_NH_1_energy,
-    >>> # NH_O_2_relidx, NH_O_2_energy, O_NH_2_relidx, O_NH_2_energy)
-    >>> dssp[a_key]
-    (3, 'A', 'H', 0.7075471698113207, -61.2, -42.4,
-     -2, -0.7, 4, -3.0, 1, -0.2, 5, -0.2)
+    How DSSP could be use::
```
Comment (peterjc) [inline Bio/PDB/DSSP.py]: used --> used

File: Bio/PDB/DSSP.py [hunk 2]
```diff
@@ -168,8 +168,12 @@ def dssp_dict_from_pdb_file(in_file, DSSP="dssp"):
 
     Examples
     --------
-    >>> dssp_dict=dssp_dict_from_pdb_file("1fat.pdb")
-    >>> aa, ss, acc=dssp_dict[('A', 1)]
+    How dssp_dict_frompdb_file could be use::
```
Comment (peterjc) [inline Bio/PDB/DSSP.py]: use --> used

File: Bio/PDB/FragmentMapper.py [hunk 3]
```diff
@@ -24,9 +24,19 @@
...
 The library files can be found in directory 'fragment_data'.
 
+    >>> from Bio.PDB.PDBParser import PDBParser
+    >>> from Bio.PDB.FragmentMapper import FragmentMapper
+    >>> parser = PDBParser()
+    >>> structure = parser.get_structure("1a8o", "PDB/1A8O.pdb")
     >>> model = structure[0]
-    >>> fm = FragmentMapper(model, lsize=10, flength=5, dir="fragment_data")
-    >>> fragment = fm[residue]
+    >>> fm = FragmentMapper(model, lsize=10, flength=5, fdir="PDB/fragment_data")
```
Comment (peterjc) [inline Bio/PDB/FragmentMapper.py]: Where is this directory? Does it get created?

I note that https://github.com/biopython/biopython/blob/master/Tests/test_PDB_FragmentMapper.py doesn't cover this argument (it probably should do).

File: Bio/PDB/PDBIO.py [hunk 4]
```diff
@@ -51,11 +51,12 @@ class PDBIO(object):
...
-    >>> p=PDBParser()
-    >>> s=p.get_structure("1fat", "1fat.pdb")
+    >>> from Bio.PDB.PDBIO import PDBIO
+    >>> parser = PDBParser()
+    >>> structure = parser.get_structure("1a8o", "PDB/1A8O.pdb")
     >>> io=PDBIO()
-    >>> io.set_structure(s)
-    >>> io.save("out.pdb")
+    >>> io.set_structure(structure)
+    >>> io.save("../../bio-pdb-pdbio-out.pdb")
```
Comment (peterjc) [inline Bio/PDB/PDBIO.py]: Writing files to random directories on the system (outside the Biopython tree is a bad idea). How about:

```
>>> io.save("out.pdb")
>>> import os
>>> os.remove("out.pdb")  # tidy up
```

This is not ideal, but we have similar cases elsewhere, e.g.
https://github.com/biopython/biopython/blob/biopython-173/Bio/SeqIO/SffIO.py#L228

File: Bio/PDB/mmcifio.py [hunk 5]
```diff
@@ -54,11 +54,12 @@ class MMCIFIO(object):
...
-        >>> p=MMCIFParser()
-        >>> s=p.get_structure("1fat", "1fat.cif")
+        >>> from Bio.PDB.mmcifio import MMCIFIO
+        >>> parser = MMCIFParser()
+        >>> structure = parser.get_structure("1a8o", "PDB/1A8O.cif")
         >>> io=MMCIFIO()
-        >>> io.set_structure(s)
-        >>> io.save("out.cif")
+        >>> io.set_structure(structure)
+        >>> io.save("../../bio-pdb-mmcifio-out.cif")
```
Comment (peterjc) [inline Bio/PDB/mmcifio.py]: Same comment as above, don't write to random folders outside the Biopython source tree, and best remove the file afterwards.

File: Bio/PDB/vectors.py [hunk 6]
```diff
@@ -139,13 +141,14 @@ def refmat(p, q):
 
     Examples
     --------
+    >>> from Bio.PDB.vectors import refmat
     >>> p,q = Vector(1, 2, 3), Vector(2, 3, 5)
```
Comment (peterjc) [inline Bio/PDB/vectors.py]: Would you put a space in, giving ``>> p, q = ...`` here please?

File: Bio/PDB/vectors.py [hunk 7]
```diff
@@ -139,13 +141,14 @@ def refmat(p, q):
...
     >>> mirror = refmat(p, q)
     >>> qq = p.left_multiply(mirror)
     >>> print(q)
     <Vector 2.00, 3.00, 5.00>
     >>> print(qq)  # q and qq should be the same
-    <Vector 1.65, 3.30, 4.94>
+    <Vector 1.21, 1.82, 3.03>
```
Comment (peterjc) [inline Bio/PDB/vectors.py]: I see what you mean about the output being very different probably the test was always wrong...

File: Bio/PDB/FragmentMapper.py (hunk 3 above)
Comment (peterjc) [reply Bio/PDB/FragmentMapper.py]: Ah, I see you are adding ``Tests/Bio/fragment_data/lib_10_z_5.txt`` to the source code.

File: Tests/PDB/fragment_data/lib_10_z_5.txt [hunk 8]
```diff
@@ -0,0 +1,131 @@
+
```
Comment (peterjc) [inline Tests/PDB/fragment_data/lib_10_z_5.txt]: See discussion on https://github.com/biopython/biopython/pull/1191#issuecomment-305338212 and https://github.com/csblab/fragments - given this is now MIT licensed, we can probably include this small file, right @JoaoRodrigues ?

Comment (peterjc) [reply Tests/PDB/fragment_data/lib_10_z_5.txt]: Actually, please reuse ``Tests/PDB/lib_10_z_5.txt`` instead (already in the repository)

**Comment by svalqui** [comment]
Ok, I think that's all feedback included.

**Pull request merged**